src/users/email_verifications/routes.rs

```diff
@@ -0,0 +1,64 @@
+use axum::extract::Path;
+use axum::routing::post;
+use axum::{extract::State, Router};
+use chrono::{Duration, Utc};
+use diesel::prelude::*;
+use diesel_async::{pooled_connection::deadpool::Pool, AsyncPgConnection, RunQueryDsl};
+use uuid::Uuid;
+
+use crate::schema::users;
+use crate::{auth::AuthExtractor, schema::email_verifications, users::models::UserRole, AppState};
+
+use super::{models::EmailVerification, EmailVerificationError};
+
+pub fn email_verification_router() -> Router<AppState> {
+    Router::new()
+        .route("/email_verification", post(create_email_verification))
+        .route("/confirm_email/:verification_id", post(confirm_email))
+}
+
+pub async fn create_email_verification(
+    auth: AuthExtractor<{ UserRole::User as u32 }>,
+    State(pool): State<Pool<AsyncPgConnection>>,
+) -> Result<(), EmailVerificationError> {
```

Comment:
```rust
pub async fn create_email_verification(
    auth: AuthExtractor<{ UserRole::User as u32 }>,
    State(pool): State<Pool<AsyncPgConnection>>,
    State(app_state): State<AppState>, // add this to get the email_username when you replace OnceCell
) -> Result<(), EmailVerificationError> {
```

> It mostly depends on the semantics of what you're working with. If your data is truly, unavoidably global in scope then make it a global. A database connection might very well qualify as such, especially if it's going to be open during the entire lifetime of the program.
On the other hand, if it's something that's only going to be shared between a few shorter-lived threads for a relatively short amount of time, then using Arcs instead would help express the more limited scope of the object's useful lifetime.

source:
https://www.reddit.com/r/rust/comments/dq65qg/lazy_static_vs_passing_arc_around/